app/controllers/client/carts_controller.rb

```diff
@@ -67,15 +67,17 @@ def check_quantity_validation product
   end
 
   def update_existed_item product
-    session[:cart].each do |key, value|
-      next unless key.to_i == product.id
-      a = value.to_i
-      b = item_params[:quantity].to_i
-      a += b
-      session[:cart][key] = a
-      remain_quantity = product.quantity - b
-      return true if product.update_attribute :quantity, remain_quantity
-      flash[:danger] = t "flash.update_fail"
+    ActiveRecord::Base.transaction do
```

Comment:
có cần thêm transaction mô chú hì, chỉ update DB 1 lần khi mà product id đã tồn tại trong cart thôi mờ, nếu chỉ update 1 record thì ko cần quá gì mình phải bật transaction nhé

transaction là 1 tập hợp các thao tác mà làm thay đổi DB
"tập hợp" nghĩa là hơn 1 thao tác làm thay đổi DB ấy nhóe

Reply:
Dạ cái ni ban đầu e cũng nghĩ như rứa. Xong a Huy có nói là mình không thể chắc chắn hoàn toàn được, hoăc sẽ có những trường hợp đặc biệt, mà ngoài product_id mình có thì cái khác pass được điều kiện thì sẽ sinh ra nhiều lần update nên cần bỏ transaction á chị.

Reply:
`mà ngoài product_id mình có thì cái khác pass được điều kiện thì sẽ sinh ra nhiều lần update `
`next unless key.to_i == product.id` 
=> chỉ có 1 record dc update chứ mấy, sao lại ko chắc chắn,em so sánh mỗi product id thôi mà, nếu trong cart em tồn tại hơn 1 giá trị product id giống nhau -> thì xử lý cart em lưu trữ bị sai chứ đúng phải là chỉ update 1 record thôi chú